src/base/sexpr.zig

```diff
@@ -0,0 +1,218 @@
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const testing = std.testing;
+
+/// Represents a node in an S-expression tree.
+pub const Node = union(enum) {
+    node: struct {
+        value: []const u8,
+        children: std.ArrayListUnmanaged(Node),
+    },
+    string: []const u8,
+    signed_int: i128,
+    unsigned_int: u128,
+    float: f64,
+
+    /// Create a new node with the given value.
+    /// The value will be duplicated so that it is owned by the node
+    /// and will be freed when the node is destroyed.
+    pub fn newNode(gpa: Allocator, value: []const u8) !Node {
```

Comment:
nit: initNode to match `init` and `deinit`

Reply:
Done in 58efb0416a